ddebug/debug.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Descent 3
+ * Copyright (C) 2024 Parallax Software
+ * Copyright (C) 2024 Descent Developers
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+
+#include "debug.h"
+
+void Debug_LogClose();
+
+FILE *Debug_logfile = nullptr;
+
+bool Debug_Logfile(const char *filename) {
```

Comment:
Should we merge this with our logging system set up at `misc/log.cpp` ?

Reply:
or did I misunderstand the purpose of this module

Reply:
Yes, it is, but our logging facility still not written, so I decided left it as is for a while.

Reply:
fair enough. How can I test this change?

Reply:
Log file is enabled by `-logfile` option. After starting logfile will be created in current directory under "d3.log" filename. Content will be identical to console output.

Reply:
file is created but empty on Linux, is that expected?

Reply:
-DLOGGER need to be enabled too as mprintf() is enabled only under  LOGGER define. This is works only in Debug build.

Reply:
yes my build is Debug and ENABLE_LOGGER is ON, investigating

Reply:
`Debug_ConsolePrintf` seems to call `Debug_LogWrite` only on Windows (winmono) and not on Linux (lnxmono)

Reply:
Confirming that it works when adding `  Debug_LogWrite(Mono_buffer);` in `Debug_ConsolePrintf` in `lnxmono.cpp`

Reply:
Seems this code was broken at first place. Added `Debug_LogWrite(Mono_buffer)` call in `Debug_ConsolePrintf` for Linux.